### DISCUSSION TITLE
Unable to create alpha shapes from point cloud

### DISCUSSION THREAD

**Discussion by schFuelPC** [isl-org/Open3D#5236] [General]
Hi all, 

Very new user of Open3D and hoping to get some assistance on the function "create_from_point_cloud_alpha_shape". Using version 15.2 of open3d for windows.

I have a point cloud (see below) of a tree canopy that I am looking to create a 3D alpha shape around to get an estimate of the volume of the canopy.  
[WC1_Cloud_canopy.txt](https://github.com/isl-org/Open3D/files/8953312/WC1_Cloud_canopy.txt)

I am using the following code to test different size alphas and then will examine the resultant meshes in CloudCompare.
```
alpha_value_list = np.arange(0, 10, 0.01)

for i, alpha in enumerate(alpha_value_list):
    pcd = o3d.io.read_point_cloud(file_name, 'xyz')
    mesh = o3d.geometry.TriangleMesh.create_from_point_cloud_alpha_shape(pcd, alpha)
    print(f'Alpha: {alpha}. Mesh: {mesh}')
    o3d.io.write_triangle_mesh(filename=f'{outpath}/{alpha}_mesh.obj',
                               mesh=mesh,
                               write_ascii=True,
                               compressed=False,
                               write_vertex_normals=True,
                               write_vertex_colors=True,
                               write_triangle_uvs=True,
                               print_progress=False)
```
    
On all values of alpha, the output of `create_from_point_cloud_alpha_shape` is `TriangleMesh with 0 points and 0 triangles`.

Any thoughts or suggestions on what I might be doing wrong?

Thanks

**Comment by adam-grant-hendry** [comment]
@schFuelPC The shape of your data is very odd. Loading it into a `numpy` array reveals it is `(2318, 11)`, which is not a set of `x, y, z` points (otherwise axis 1 would be 3 instead of 11). What is this data of?

The error `TriangleMesh with 0 points and 0 triangles` usually happens when `alpha` is not set big enough, but it may also be in your case that the dataset does not actually appear to be real points.

**NOTE:** The [`open3d` examples](http://www.open3d.org/docs/latest/tutorial/Advanced/surface_reconstruction.html) use very small alpha values, which erroneously leads many users to think their alpha values should be small too. Alpha must be at least a little bigger than the the smallest distance between three points in your mesh. The choice of alpha depends on how spread out points are in your dataset.

If I use a large alpha like `60000`, I get a triangle mesh (the smallest number that yields some triangles is about `1000`). But, as I suspected, the output doesn't make much sense:

![image](https://user-images.githubusercontent.com/59346180/200102313-9b1832ea-c7f2-44d3-94a7-71b0b1c166fb.png)

I would suggest reexamining your input data set and first try to organize it into an `N x 3` shape of 3d data points.
